Added Gen 8 Pokemon Compatibility

deadaccountld opened the pull request.

deadaccountld committed:
Update pokedex.h

Added Pokemon and alternate forms introduced in Generation 8. Does not include Sword and Shield DLC as information has yet to be revealed to be practical at this time.

deadaccountld committed:
Update Pokedex_Data_Table.c

deadaccountld committed:
Update Pokedex_Data_Table.c

Added Entries for Grookey-Eternatus + fixed some category name errors

deadaccountld committed:
Update Pokedex_Data.string

deadaccountld committed:
Update pokedex.h

deadaccountld committed:
Update Pokedex_Data_Table.c

deadaccountld committed:
Update pokedex.h

deadaccountld committed:
Update Pokedex_Data.string

deadaccountld committed:
Update Pokemon_Name_Table.string

deadaccountld committed:
Update Pokedex_Data.string

deadaccountld committed:
Update Pokedex_Data_Table.c

deadaccountld committed:
Update Species_To_Pokdex_Table.c

deadaccountld committed:
Update Species_To_Pokdex_Table.c

deadaccountld committed:
Update pokedex.h

deadaccountld committed:
Update Pokedex_Orders.c

deadaccountld committed:
Update pokedex.h

deadaccountld committed:
Update species.h

deadaccountld committed:
Update Pokedex_Orders.c

deadaccountld committed:
Update Pokedex_Data.string

deadaccountld committed:
Update Habitat_Table.c

deadaccountld committed:
Update items.h

deadaccountld committed:
Update Back_Pic_Coords_Table.c

deadaccountld committed:
Update Cry_Table.c

deadaccountld committed:
Update Cry_Table_2.c

deadaccountld committed:
Update Enemy_Elevation_Table.c

deadaccountld committed:
Update Footprint_Table.c

deadaccountld committed:
Update Icon_Palette_Table.c

deadaccountld committed:
Update Item_Animation_Table.c

deadaccountld committed:
Update Front_Pic_Table.c

deadaccountld committed:
Update Front_Pic_Coords_Table.c

deadaccountld committed:
Update Palette_Table.c

deadaccountld committed:
Update Back_Pic_Table.c

deadaccountld committed:
Update Shiny_Palette_Table.c

deadaccountld committed:
Update Icon_Table.c

deadaccountld committed:
Update Egg_Moves.c

deadaccountld committed:
Update Evolution Table.c

deadaccountld committed:
Update Evolution Table.c

Did not include Gigantamax Forms

deadaccountld committed:
Update items.h

Added new evolution items from Sword and Shield

deadaccountld committed:
Update sprite_data.h

deadaccountld committed:
Update cry_data.h

deadaccountld committed:
Update Back_Pic_Coords_Table.c

deadaccountld committed:
Update Cry_Table.c

deadaccountld committed:
Update cry_data.h

deadaccountld committed:
Update Enemy_Elevation_Table.c

deadaccountld committed:
Update Cry_Table_2.c

deadaccountld committed:
Update Footprint_Table.c

deadaccountld committed:
Update Front_Pic_Coords_Table.c

deadaccountld committed:
Update Icon_Palette_Table.c

deadaccountld committed:
Update Item_Animation_Table.c

deadaccountld committed:
Update Base_Stats.c

deadaccountld committed:
Update Base_Stats.c

deadaccountld committed:
ADDED GEN 8

deadaccountld committed:
Update Base_Stats.c

Added Frosmoth

deadaccountld committed:
Update Learnsets.c

deadaccountld committed:
Update Learnsets.c

deadaccountld committed:
Update Learnsets.c

deadaccountld committed:
Update items.h

Added Rusted Sword and Rusted Shield

deadaccountld committed:
Update species.h

Added Gigantamax form for Low Key Toxtricity due to differing moves/abilities

deadaccountld committed:
Update sprite_data.h

Defined Toxtricity Low Key Gigantamax Form

deadaccountld committed:
Update Pokemon_Name_Table.string

Defined Gigantamax form for Low Key Toxtricity

deadaccountld committed:
Update Back_Pic_Coords_Table.c

Defined Low Key GMax Toxtricity

deadaccountld committed:
Update Back_Pic_Table.c

deadaccountld committed:
Update Base_Stats.c

deadaccountld committed:
Update Cry_Table.c

deadaccountld committed:
Update Cry_Table_2.c

deadaccountld committed:
Update Enemy_Elevation_Table.c

deadaccountld committed:
Update Footprint_Table.c

deadaccountld committed:
Update Front_Pic_Coords_Table.c

deadaccountld committed:
Update Front_Pic_Table.c

deadaccountld committed:
Update Icon_Palette_Table.c

deadaccountld committed:
Update Icon_Table.c

deadaccountld committed:
Update Item_Animation_Table.c

deadaccountld committed:
Update Palette_Table.c

deadaccountld committed:
Update Back_Pic_Table.c

deadaccountld committed:
Update Front_Pic_Table.c

deadaccountld committed:
Update Shiny_Palette_Table.c

deadaccountld committed:
Update Species_To_Pokdex_Table.c

deadaccountld committed:
Update Learnsets.c

deadaccountld committed:
Update Learnsets.c

Skeli789 reviewed (changes requested):
-Don't remove alternate forms from the Pokedex tables (like Wormadam and Rotom forms)
-Comment out entries referencing non-existent sprites and cries
-Keep habitats in same general order (eg. don't put Skwovet at the beginning of Forest; it should be on the first new Forest page)
-Make sure habitats have ALL alternate forms (except for Megas and Gigas)
-Put SPECIES_UNKNOWN_MYTHICAL back after Zarude. It doesn't need to added to the data tables, but datamining showed that there is another mythical that will be released. It's best if we can anticipate this to cause people who decide to go ahead with Gen 8 the least amount of problems
-The species name table MUST be in species order. Eg. Grookey should be the first new entry at the end, and not placed after Marshadow
-The Pokedex species defines (NATIONAL_DEX_) should be based on the Pokemon name, and not include forms (so no NATIONAL_DEX_URSHIFU_SINGLE)
-We're just calling regular Sinistea SPECIES_SINISTEA, and not SPECIES_SINISTEA_ANTIQUE
-Add new defines for the new evolution methods (like Nature based, Critical Hit, etc.). I'll implement them in the CFRU when they're more relevant
-Make the Rusted Sword and Shield's item defines to be immediately after the other hold items (0x2C6 and 0x2C7). The evolution items should follow after that
-Make sure it compiles

deadaccountld committed:
Update Pokedex_Orders.c

Reinserted the deleted alternate forms

deadaccountld committed:
Update Back_Pic_Coords_Table.c

Commented out Gen 8

deadaccountld committed:
Update Back_Pic_Table.c

Commented out Gen 8

deadaccountld committed:
Update Cry_Table.c

Commented out Gen 8

deadaccountld committed:
Update Cry_Table_2.c

Commented out Gen 8

deadaccountld committed:
Update Enemy_Elevation_Table.c

Commented out Gen 8

deadaccountld committed:
Update Front_Pic_Coords_Table.c

Commented out Gen 8

deadaccountld committed:
Update Front_Pic_Table.c

Commented out Gen 8

deadaccountld committed:
Update cry_data.h

Commented out Gen 8

deadaccountld committed:
Update sprite_data.h

Commented out Gen 8

deadaccountld committed:
Update species.h

Readded unknown mythical

deadaccountld committed:
Update Pokemon_Name_Table.string

Moved all Gen 8 entries to the end of the table

deadaccountld committed:
Update pokedex.h

Removed form names in #define NATIONAL_DEX_'s

deadaccountld committed:
Update items.h

Made Rusted Sword and Shield follow hold items followed by evolution items

deadaccountld committed:
Update evolution.h

Added Sirfetch'd, Toxtricity and Runerigus evolution methods

deadaccountld committed:
Update evolution.h

deadaccountld committed:
Update Evolution Table.c

Commented out pokemon that lack info + changed Yamask_G, Farfetch'd-G and Toxel's Evo methods

deadaccountld committed:
Update Habitat_Table.c

Removed clone of Tirtouga line, added Snorlax line, added Gen 8 + Galarian forms

deadaccountld committed:
Update Pokedex_Data.string

Commented out Pokemon with no dex entries revealed yet

deadaccountld committed:
Update Species_To_Pokdex_Table.c

deadaccountld committed:
Update Pokedex_Data.string

deadaccountld committed:
Update Shiny_Palette_Table.c

deadaccountld committed:
Update Pokedex_Data_Table.c

deadaccountld committed:
Update Palette_Table.c

deadaccountld committed:
Update Item_Animation_Table.c

deadaccountld committed:
Update Learnsets.c

deadaccountld committed:
Update Icon_Table.c

deadaccountld committed:
Update Icon_Palette_Table.c

deadaccountld committed:
Update Base_Stats.c

deadaccountld committed:
Update sprite_data.h

deadaccountld committed:
Update Base_Stats.c

deadaccountld committed:
Update Base_Stats.c

deadaccountld committed:
Update Egg_Moves.c

deadaccountld committed:
Update Evolution Table.c

deadaccountld committed:
Update Evolution Table.c

deadaccountld committed:
Update Habitat_Table.c

deadaccountld committed:
Update Learnsets.c

deadaccountld committed:
Update pokedex.h

deadaccountld committed:
Update Pokedex_Data_Table.c

deadaccountld committed:
Update Pokedex_Data_Table.c

deadaccountld committed:
Update pokedex.h

deadaccountld committed:
Update Pokedex_Orders.c

deadaccountld committed:
Update Pokedex_Orders.c

deadaccountld committed:
Update Species_To_Pokdex_Table.c

deadaccountld committed:
Update Base_Stats.c

deadaccountld committed:
Update Pokedex_Data.string

deadaccountld committed:
Update Pokedex_Data.string

deadaccountld committed:
Update Pokemon_Name_Table.string

deadaccountld committed:
Update Back_Pic_Coords_Table.c

deadaccountld committed:
Update Front_Pic_Coords_Table.c
